Code/GraphMol/MolDraw2D/MolDraw2D.cpp

```diff
@@ -544,19 +544,22 @@ void MolDraw2D::calculateScale(int width, int height) {
 
   x_range_ = x_max - x_min_;
   y_range_ = y_max - y_min_;
-  if (x_range_ > 1e-4 && y_range_ > 1e-4) {
-    scale_ = std::min(double(width) / x_range_, double(height) / y_range_);
-  } else {
-    scale_ = 0;
+  if (x_range_ < 1e-4) {
+    x_range_ = 1.;
+    x_min_ -= 0.5;
+    x_max += 0.5;
   }
-
-  // std::cerr << "  " << x_max << "-" << x_min_ << " " << x_range_ << "    "
-  //           << y_max << "-" << y_min_ << " " << y_range_ << std::endl;
+  if (y_range_ < 1e-4) {
+    y_range_ = 1.;
+    y_min_ -= 0.5;
+    y_max += 0.5;
+  }
+  scale_ = std::min(double(width) / x_range_, double(height) / y_range_);
 
   // we may need to adjust the scale if there are atom symbols that go off
   // the edges, and we probably need to do it iteratively because
   // get_string_size uses the current value of scale_.
-  while (scale_ > 0.0) {
+  while (scale_ > 1e-4) {
```

Comment:
These types of while loops are suspicious to me, are we guaranteed to terminate?  Looks like if the scale doesn't change we might be.